src/components/Modals/ContactCardModal.jsx

```diff
@@ -62,84 +62,22 @@ class ContactCardModal extends React.Component {
-                          <ModalHeader className="u-flex u-flex-items-center u-flex-column-s u-pr-1-half-s">
-                            {children}
-                            <div className="u-flex u-flex-row u-ml-auto u-ml-0-s">
-                              <ContactGroups
-                                contact={contact}
-                                allGroups={allGroups}
-                              />
-                              <ContactCardMenu
-                                deleteAction={{
-                                  label: t('delete'),
-                                  action: this.toggleConfirmDeleteModal
-                                }}
-                                editAction={{
-                                  label: t('edit'),
-                                  action: this.toggleEditMode
-                                }}
-                              />
-                            </div>
-                          </ModalHeader>
-                        )}
-                        renderBody={children => (
-                          <ModalContent>{children}</ModalContent>
-                        )}
-                      />
-                    )
-                  }
-
-                  if (editMode && fetchStatusContact === 'loaded') {
-                    return (
-                      <ContactFormModal
-                        contact={contact}
-                        onClose={this.toggleEditMode}
-                        title={t('edit-contact')}
-                        afterMutation={this.toggleEditMode}
-                      />
-                    )
-                  }
-
-                  if (shouldDisplayConfirmDeleteModal) {
-                    return (
-                      <Modal
-                        into="body"
-                        title={t('delete-confirmation.title', {
-                          smart_count: 1
-                        })}
-                        description={t(
-                          getConnectedAccounts(contact).length > 0
-                            ? 'delete-confirmation.description-google'
-                            : 'delete-confirmation.description-simple',
-                          {
-                            smart_count: 1
-                          }
-                        )}
-                        primaryText={t('delete')}
-                        primaryType="danger"
-                        primaryAction={() => this.deleteContact(contact)}
-                        secondaryText={t('cancel')}
-                        secondaryAction={this.toggleConfirmDeleteModal}
-                        dismissAction={this.toggleConfirmDeleteModal}
-                      />
-                    )
-                  }
+                  return (
+                    <DumpContactCardModal
+                      fetchStatusContact={fetchStatusContact}
+                      allGroupsContactStatus={allGroupsContactStatus}
```

Comment:
AMHA we should let the componnet creating the Query be responsible to show the spinner if the data is not loaded. This would remove the need of the fetchStatusContact (which could maybe be named contactFetchStatus) property and would keep the query responsibility in a single component. This would also make DumbContactCardModal (careful, the component is named DumpContactCardModal here; dump -> dumb) easier to test as the test would not have to focus on the query side.

Reply:
Complètement d'accord. J'y suis juste allé comme un vrai bouché en déplaçant tout le composant dans un premier temps. Je voulais aussi le simplifier avec des useQuery() qui le rendrait bien plus lisible, mais t'as raison pour les fetch c'est rapide à faire.